types/begin__data/begin__data-tests.ts

```diff
@@ -0,0 +1,397 @@
+import data, { BeginDataType, ReadonlyBeginDataType } from '@begin/data';
+
+interface TestData {
+    key: string;
+    name?: string;
+    age?: number;
+}
+
+const table = 'test-table';
+const key = 'test-key';
+
+const obj: BeginDataType = {
+    propString: 'data',
+    propNumber: 1.23,
+    propBoolean: false,
+    propArrayEmpty: [],
+    propObjectEmpty: {},
+    propArray: ['value', 0, true],
+    propObject: {
+        propNestedObject: {
+            propNestedArray: [{}, 1],
+        },
+        propNestedString: 'value',
+    },
+};
+// $ExpectType ReadonlyBeginDataType
+obj as ReadonlyBeginDataType;
+
+// Get without key.
+(async () => {
+    const res = await data.get({ table });
+    // $ExpectType DataGetEntireTableResult
+    res;
+    // $ExpectType string
+    res[0].table;
+    res[0].table = 'another-table';
+    // $ExpectType string
+    res[0].key;
+    // $ExpectType TestData
+    res[0] as TestData;
+    const cursor = res.cursor;
+    if (typeof cursor === 'string') {
+        // $ExpectType string
+        cursor;
+    } else {
```

Comment:
```suggestion
    if (typeof cursor !== 'string') {
```

Reply:
I rewrote them with single line `// $ExpectType string | undefined` that is the same in the section below that.